api/migrations/0003_fake_data.py

```diff
@@ -152,29 +190,86 @@ def _make_teacher_under_18_recipe(self):
             is_validated=self.faker.pybool,
         )
 
-    def _make_fake_coordinators(self):
-        """Makes fake coordinators."""
-        self.coordinator_recipe.make(_quantity=AMOUNT_OF_COORDINATORS)
+    def __get_random_time_or_none(self) -> time | None:
+        return self.faker.random_element([None, self.faker.time_object()])
 
-    def _make_fake_students(self):
-        """Makes fake students."""
-        self.student_recipe.make(_quantity=AMOUNT_OF_STUDENTS)
+    def _make_group_recipe(self) -> Recipe:
+        group_common_recipe = self._make_group_common_recipe(
+            APP_NAME + ".Group", teachers_min_value=1, teachers_max_value=3
+        )
+        return group_common_recipe.extend(
+            availability_slot=lambda: self._get_random_amount_of_objects(
+                DayAndTimeSlot, min_length=10, max_length=20
+            ),
+            is_for_staff_only=self.faker.pybool,
+            language_and_level=lambda: self.faker.random_element(
+                LanguageAndLevel.objects.all()
+            ),
+            lesson_duration_in_minutes=lambda: self.faker.pyint(
+                min_value=30, max_value=120, step=30
+            ),
+            status=lambda: self.faker.random_element(GroupStatus.values),
+            start_date=self.faker.past_date,
+            end_date=self.faker.future_date,
+            # FIXME: Sometimes all of the days are generated as None.
+            #  It violates the model's constraint.
+            #  Just restart the migration and pray to the God of Random.
+            monday=self.__get_random_time_or_none,
+            tuesday=self.__get_random_time_or_none,
+            wednesday=self.__get_random_time_or_none,
+            thursday=self.__get_random_time_or_none,
+            friday=self.__get_random_time_or_none,
+            saturday=self.__get_random_time_or_none,
+            sunday=self.__get_random_time_or_none,
+        )
 
-    def _make_fake_teachers(self):
-        """Makes fake teachers."""
-        self.teacher_recipe.make(_quantity=AMOUNT_OF_TEACHERS)
+    def _make_speaking_club_recipe(self) -> Recipe:
+        group_common_recipe = self._make_group_common_recipe(
+            APP_NAME + ".SpeakingClub", teachers_min_value=0, teachers_max_value=2
+        )
+        return group_common_recipe.extend(
+            is_for_children=self.faker.pybool,
+            language=lambda: self.faker.random_element(Language.objects.all()),
+            teachers_under_18=lambda: related(
+                *[self.teacher_under_18_recipe]
+                * self.faker.pyint(min_value=0, max_value=2)
+            ),
+        )
+
+    def _make_fake_coordinators_without_group(self):
+        """Makes fake coordinators without group."""
+        self.coordinator_recipe.make(_quantity=AMOUNT_OF_COORDINATORS_WITHOUT_GROUP)
```

Comment:
looks a bit inconsequential that you're using constants here but plain numbers in `min_value` and `max_value`

Reply:
Fixed it in places that could be changed